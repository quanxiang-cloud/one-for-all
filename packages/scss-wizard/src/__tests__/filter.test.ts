jest.mock('../format');
import fs from 'fs';
import path from 'path';
import toAST from '../to-ast';
import toSCSS from '../to-scss';

test('toAST_should_throw_if_encounter_bad_scss', async () => {
  const bemSCSS = fs.readFileSync(path.join(__dirname, 'fixtures/bem.scss'),  { encoding: 'utf-8' });
  const ast = await toAST(bemSCSS);

  const scss = await toSCSS(ast);
  console.log(scss);

  expect(ast.nodes).toMatchSnapshot();
});
