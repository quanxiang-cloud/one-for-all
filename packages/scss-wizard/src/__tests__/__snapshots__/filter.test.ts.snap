// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toAST_should_throw_if_encounter_bad_scss 1`] = `
Array [
  Object {
    "indexes": Object {},
    "lastEach": 8,
    "nodes": Array [
      Object {
        "initialIndex": 2,
        "name": "include",
        "params": "someMixin",
        "position": 4,
        "raws": Object {
          "afterName": " ",
          "before": "

  ",
          "between": "",
        },
        "source": Object {
          "end": Object {
            "column": 21,
            "line": 8,
            "offset": 135,
          },
          "inputId": 0,
          "start": Object {
            "column": 3,
            "line": 8,
            "offset": 117,
          },
        },
        "type": "atrule",
      },
      Object {
        "initialIndex": 1,
        "position": 5,
        "prop": "display",
        "raws": Object {
          "before": "
  ",
          "between": ":",
        },
        "sortProperty": true,
        "source": Object {
          "end": Object {
            "column": 18,
            "line": 6,
            "offset": 112,
          },
          "inputId": 0,
          "start": Object {
            "column": 3,
            "line": 6,
            "offset": 97,
          },
        },
        "type": "decl",
        "value": "inline",
      },
      Object {
        "initialIndex": 0,
        "position": 5,
        "prop": "font-size",
        "raws": Object {
          "before": "
  ",
          "between": ":",
        },
        "sortProperty": true,
        "source": Object {
          "end": Object {
            "column": 18,
            "line": 5,
            "offset": 93,
          },
          "inputId": 0,
          "start": Object {
            "column": 3,
            "line": 5,
            "offset": 78,
          },
        },
        "type": "decl",
        "value": "40px",
      },
      Object {
        "indexes": Object {},
        "initialIndex": 5,
        "lastEach": 8,
        "nodes": Array [
          Object {
            "initialIndex": 0,
            "position": 5,
            "prop": "color",
            "raws": Object {
              "before": "
    ",
              "between": ":",
            },
            "sortProperty": true,
            "source": Object {
              "end": Object {
                "column": 16,
                "line": 19,
                "offset": 254,
              },
              "inputId": 0,
              "start": Object {
                "column": 5,
                "line": 19,
                "offset": 243,
              },
            },
            "type": "decl",
            "value": "blue",
          },
        ],
        "position": 6,
        "raws": Object {
          "after": "
  ",
          "before": "

  ",
          "between": "",
          "semicolon": true,
        },
        "selector": "&:hover",
        "source": Object {
          "end": Object {
            "column": 3,
            "line": 20,
            "offset": 258,
          },
          "inputId": 0,
          "start": Object {
            "column": 3,
            "line": 18,
            "offset": 229,
          },
        },
        "type": "rule",
      },
      Object {
        "indexes": Object {},
        "initialIndex": 4,
        "lastEach": 8,
        "nodes": Array [
          Object {
            "initialIndex": 0,
            "position": 5,
            "prop": "float",
            "raws": Object {
              "before": "
    ",
              "between": ":",
            },
            "sortProperty": true,
            "source": Object {
              "end": Object {
                "column": 17,
                "line": 15,
                "offset": 220,
              },
              "inputId": 0,
              "start": Object {
                "column": 5,
                "line": 15,
                "offset": 208,
              },
            },
            "type": "decl",
            "value": "right",
          },
        ],
        "position": 7,
        "raws": Object {
          "after": "
  ",
          "before": "

  ",
          "between": "",
          "semicolon": true,
        },
        "selector": ".icon",
        "source": Object {
          "end": Object {
            "column": 3,
            "line": 16,
            "offset": 224,
          },
          "inputId": 0,
          "start": Object {
            "column": 3,
            "line": 14,
            "offset": 196,
          },
        },
        "type": "rule",
      },
      Object {
        "indexes": Object {},
        "initialIndex": 3,
        "lastEach": 8,
        "name": "media",
        "nodes": Array [
          Object {
            "initialIndex": 0,
            "position": 5,
            "prop": "background-color",
            "raws": Object {
              "before": "
    ",
              "between": ":",
            },
            "sortProperty": true,
            "source": Object {
              "end": Object {
                "column": 32,
                "line": 11,
                "offset": 187,
              },
              "inputId": 0,
              "start": Object {
                "column": 5,
                "line": 11,
                "offset": 160,
              },
            },
            "type": "decl",
            "value": "aliceblue",
          },
        ],
        "params": "screen",
        "position": 8,
        "raws": Object {
          "after": "
  ",
          "afterName": " ",
          "before": "

  ",
          "between": "",
          "semicolon": true,
        },
        "source": Object {
          "end": Object {
            "column": 3,
            "line": 12,
            "offset": 191,
          },
          "inputId": 0,
          "start": Object {
            "column": 3,
            "line": 10,
            "offset": 140,
          },
        },
        "type": "atrule",
      },
    ],
    "raws": Object {
      "after": "
",
      "before": "",
      "between": "",
      "semicolon": true,
    },
    "selector": ".button",
    "source": Object {
      "end": Object {
        "column": 1,
        "line": 21,
        "offset": 260,
      },
      "inputId": 0,
      "start": Object {
        "column": 1,
        "line": 4,
        "offset": 66,
      },
    },
    "type": "rule",
  },
  Object {
    "indexes": Object {},
    "lastEach": 8,
    "nodes": Array [
      Object {
        "initialIndex": 0,
        "position": 5,
        "prop": "background-color",
        "raws": Object {
          "before": "
  ",
          "between": ":",
        },
        "sortProperty": true,
        "source": Object {
          "end": Object {
            "column": 24,
            "line": 24,
            "offset": 304,
          },
          "inputId": 0,
          "start": Object {
            "column": 3,
            "line": 24,
            "offset": 283,
          },
        },
        "type": "decl",
        "value": "red",
      },
    ],
    "raws": Object {
      "after": "
",
      "before": "

",
      "between": "",
      "semicolon": true,
    },
    "selector": ".button--danger",
    "source": Object {
      "end": Object {
        "column": 1,
        "line": 25,
        "offset": 306,
      },
      "inputId": 0,
      "start": Object {
        "column": 1,
        "line": 23,
        "offset": 263,
      },
    },
    "type": "rule",
  },
]
`;
